### PR TITLE
Consider bumping the minimum Python version

### PR DESCRIPTION
**Pull request by jeremyh**
This PR may be controversial, but I noticed our Docker container now uses Python 3.8, and hence all of our tests are [now only running on Python 3.8](https://github.com/opendatacube/datacube-core/runs/3356579182?check_suite_focus=true#step:8:34) (from what I can see?).

- (Edit: there's also a workflow I missed that does a conda install to Python 3.7, but it doesn't run the test suite.)

My experience is that it's very easy to break 3.6 support by accident if not testing with it. And we're increasingly seeing dependencies drop support for 3.6 on other repositories in the ODC ecosystem.

Some options:

1. *A recommendation of 3.8*. The first commit updates all docs to recommend and default to Python 3.8, but setup.py still allows Python 3.6 installs.
2. *A hard requirement on 3.8*: the second commit requires Python 3.8+ in setup.py too, so users cannot accidentally pip install newer versions on 3.6. (edit: and to bump conda smoke test version)

I'm happy to remove the second commit if people think it's too harsh. But I think a minimum requirement is friendlier to users than exposing people to unexpected bugs. 

Thoughts?

ps. The other option, of course, it to make 3.7 the minimum. But is that still common anywhere, among non-Docker users? And do we want to test with it?
pps. I also bumped the Postgres version recommendation, since none of us test on 9.5. But there's no hard requirement.

### CODE REVIEW

**Comment by snowman2** [comment]
For reference on making Python 3.7 minimum supported version: https://github.com/pyproj4/pyproj/issues/790

File: conda-environment.yml [hunk 1]
```diff
@@ -4,15 +4,15 @@ channels:
...
+  - python >=3.8
   - setuptools
   - setuptools_scm >=3.4
   - toml
   - pytest-runner
   - sphinx
   - sphinx-click
   - click
-  - python >=3.6
+  - python >=3.8
```
Comment (Kirill888) [inline conda-environment.yml]: duplicate?

**Review by Kirill888 — APPROVED** [review]
I have no issues with this change.

We currently run tests with 3.8 only, but I guess we "test" 3.6 in sandbox, which will hopefully soon switch to 3.8 too.

**Review by omad — APPROVED** [review]
I'm fine with deprecating 3.6 support and skipping 3.7.

NCI and DEA Sandbox are both still running 3.6, but are both long overdue for an update.

**Comment by omad** [comment]
It's also good timing soon after a release. We should bump to 1.9.0 for the next release.

File: conda-environment.yml (hunk 1 above)
Comment (jeremyh) [reply conda-environment.yml]: Looks like it. Removed in 96bcddc0bd0beff4419a48756e1953b9f430bf5e

**Comment by Kirill888** [comment]
@woodcockr EASI is using Python 3.8 already isn't?

**Comment by alexgleith** [comment]
Sandbox images are now working in DE Africa with Python 3.8 and it seems fine.

**Comment by jeremyh** [comment]
> NCI and DEA Sandbox are both still running 3.6, but are both long overdue for an update.

I think part of the previous difficulty was because the DEA code had a dependency that was strictly python3.6-only. I have a PR to move it: https://github.com/GeoscienceAustralia/digitalearthau/pull/305

**Comment by woodcockr** [comment]
@Kirill888 Yep, EASI is on Python 3.8 already and everything we use it working fine. Thank you for asking.

**Review by uchchwhash — APPROVED** [review]

**Pull request merged**